Cython/Compiler/FlowControl.py

```diff
@@ -677,6 +679,14 @@ def visit_CascadedAssignmentNode(self, node):
 
 class ControlFlowAnalysis(CythonTransform):
 
+    def find_in_stack(self, env):
+        if env == self.env:
+            return self.flow
+        for e, flow in reversed(self.stack):
+            if e==env:
```

Comment:
You can probably also use `is` instead of `==` for these comparisons. Scopes should never compare equal unless they are identical.
```suggestion
            if e == env:
```

Reply:
I edited your suggestion to propose "is" instead then (it seemed the easiest way to make the change quickly)